xapian-letor/configure.ac

```diff
@@ -385,6 +385,12 @@ AH_BOTTOM(
 ])
 
 AC_DEFINE([XAPIAN_LIB_BUILD], [1], [Workaround for avoiding sub-header included error])
+AC_DEFINE([BUILD_BACKEND_GLASS], [1], [Workaround for avoiding setting the conditionals])
+AC_DEFINE([BUILD_BACKEND_INMEMORY], [1], [Workaround for avoiding setting the conditionals])
+AC_DEFINE([BUILD_BACKEND_REMOTE], [1], [Workaround for avoiding setting the conditionals])
```

Comment:
These aren't preprocessor symbols; they're only used as conditionals inside the autoconf system. I think I perhaps led you in slightly the wrong direction here, however — apologies. I think you want `AM_CONDITIONAL` instead, but @ojwb is better on the autotools system than I am. (I do recommend reading the introduction to the autoconf and automake manuals so you understand how they fit together with the rest of the build system.)